app/controllers/forest_liana/sessions_controller.rb

```diff
@@ -1,108 +1,80 @@
 module ForestLiana
   class SessionsController < ForestLiana::BaseController
-    def create
+    def login_with_password
       @error_message = nil
-      @user_class = ForestLiana.user_class_name.constantize rescue nil
-
-      user = check_user
-      token = encode_token(user) if user
+      rendering_id = params['renderingId']
+      project_id = params['projectId']
+      email = params['email']
+      password = params['password']
+      two_factor_token = params['token']
+      two_factor_registration = params['twoFactorRegistration']
 
-      if token
-        render json: { token: token }, serializer: nil
-      else
-        if @error_message
-          render serializer: nil, json: JSONAPI::Serializer.serialize_errors(
-            [{ detail: @error_message }]), status: :unauthorized
-        elsif !has_internal_authentication? && ForestLiana.allowed_users.empty?
-          render serializer: nil, json: JSONAPI::Serializer.serialize_errors(
-            [{ detail: 'Forest cannot retrieve any users for the project ' \
-              'you\'re trying to unlock.' }]), status: :unauthorized
-        else
-          head :unauthorized
-        end
-      end
+      process_login(
+        use_google_authentication: false,
+        rendering_id: rendering_id,
+        project_id: project_id,
+        auth_data: { :email => email, :password => password },
```

Comment:
'email' or :email is required here to do not have the value of the varialbe as key. I changed it because I fell into that trap.

```
$ irb                                                                                                                                                                                                 
2.3.4 :001 > email = 'arnaud'
 => "arnaud"
2.3.4 :002 > { email => email }
 => {"arnaud"=>"arnaud"}
```

Reply:
<img width="221" alt="screen shot 2018-09-11 at 07 07 52" src="https://user-images.githubusercontent.com/664261/45339164-773a6080-b591-11e8-94d4-0afa8e84e3e9.png">

`foo:` is just a simpler syntax for `:foo =>`